tests/phantom/tools.js

```diff
@@ -0,0 +1,47 @@
+import {expect} from 'chai';
+import PhantomEnvironment from '../../lib/PhantomEnvironment';
+import Parser from '../../lib/Parser';
+import http from 'http';
+
+const port = 60053;
+const url = `http://localhost:${port}`;
+let responseHtml = '';
+
+export function createParser () {
+    return new Parser({
+        environment: new PhantomEnvironment({url})
+    });
+}
+
+export function createServer () {
+    return new Promise((resolve, reject) => {
+        const server = http.createServer((request, response) => {
+            response.statusCode = 200;
+            response.end(`
+                <html>
+                <head>
+                    <meta charset="utf-8">
+                </head>
+                <body>
+                ${responseHtml}
+                </body>
+                </html>
+            `);
+        });
+
+        server.listen(port, () => resolve());
+    });
+}
+
+export function setServerResponse (html, fn) {
+    responseHtml = html;
+
+    if (fn) {
+        responseHtml += `
+            <script>
+                ${fn.toString()}
+                run();
```

Comment:
Look, sometime you need a script on the page. For example, to check that `click` action works well, you need a page which react on click somehow. In this test you can set server response like this:

```js
setServerResponse(
    `
        <div>text</div>
    `,
    function run () {
        document.getElementsByTagName('DIV')[0].addEventListener('click', function(e) {
            e.target.innerHTML = 'clicked';
        })
    }
);
```

Every function's (except native functions) `toString` method returns source code of the function. That's what we do on line 42 - just injecting the function code into the server response. Right after that we invoke it with `run()`, that's why all the functions in tests has this name. But it can be easily replaced with `(${fn.toString()})();`